### DISCUSSION TITLE
Dwsim Automation

### DISCUSSION THREAD

**Discussion by ghost** [DanWBR/dwsim#412] [Q&A]
Hi, I am a student and interested in learning how to use the DWSIM Automation. I understand that automating DWSIM allow us to manipulate the flowsheets and run optimization directly from Excel, is it means that we can import the real time data from excel to the simulation and export it to excel from the simulation? If yes, how can I do that?

I have also tried to do the sample automation (Cavett's problem) using Excel VBA, but I am looking for clearer steps like how to add reference to CAPE-OPEN 1.1 Type Library, DWSIM Simulator Automation Interface and DWSIM Simulator Interface Definitions Library.

Thanks in advance, hope anyone here can assist me

**Comment by DanWBR** [comment]
@W1lfr3d99 I recommend you to try automation using Python instead. Using Excel VBA is too painful.

See this: https://gist.github.com/DanWBR/c355fd5420d20d960f5d084a7142cde8

Install pythonnet v2.x on your python distro.

**Reply by ghost** [reply]
Hi Dan,
Thanks for your recommendation. I am trying to use Python to import the value from Excel to run the simulation in DWSIM and extract the result from DWSIM to Excel. Would be very grateful if you can guide me if its possible.

**Reply by DanWBR** [reply]
it is not something that can be done quickly. try to find a python+excel sample on the internet and then mix the code with the one I've posted above.. it should be enough to get started.

**Reply by ghost** [reply]
Hi Dan,
I am trying to run this on Jupyter Notebook and stuck at line 11 (https://gist.github.com/DanWBR/c355fd5420d20d960f5d084a7142cde8)

Below is the error shown, do you know how can i solve this?

FileNotFoundException       Traceback (most recent call last)
~\AppData\Local\Temp\ipykernel_17732\1166761962.py in <module>
----> 1 clr.AddReference(dwsimpath + "CapeOpen.dll"

FileNotFoundException: Unable to find assembly 'C:\DWSIMCapeOpen.dll'.
at Python.Runtime.CLRModule.AddReference(String name)
